test/unit/test_configcommon.py

```diff
@@ -124,3 +125,38 @@ def test_clean_dynamic_variables(autosubmit_config: Callable) -> None:
 
     assert len(as_conf.dynamic_variables) == 1
     assert 'jaspion_eats' in as_conf.dynamic_variables
+
+
+def test_yaml_deprecation_warning(tmp_path, autosubmit_config: Callable):
+    """Test that the conversion from YAML to INI works as expected, without warnings.
+
+    +    Creates a dummy AS3 INI file, calls ``AutosubmitConfig.ini_to_yaml``, and
+    +    verifies that the YAML files exists and is not empty, and a backup file was
+    +    created. All without warnings being raised (i.e. they were suppressed).
+    +    """
+
+    as_conf: AutosubmitConfig = autosubmit_config(expid='a000', experiment_data={})
+    ini_file = tmp_path / 'a000_jobs.ini'
+    with open(ini_file, 'w+') as f:
+        f.write(dedent('''\
+            [LOCAL_SETUP]
+            FILE = LOCAL_SETUP.sh
+            PLATFORM = LOCAL
+            '''))
+        f.flush()
+    as_conf.ini_to_yaml(root_dir=tmp_path, ini_file=str(ini_file))
+
+    print(f'tmp_path: {tmp_path}')
```

Comment:
You should not print in tests. It is bad practice, and in fact pytest won't output it, as pytest captures the stdout/stderr (you can tell it to output, but in some corner cases that doesn't work as expected).

If you use this for writing your tests, just remember to delete it before pushing or before assigning a reviewer.